### PR TITLE
Failure Handling

### PR DESCRIPTION
**Pull request by abergfeld**
Connects to https://github.com/Banno/Grip2-Android/issues/671

Add ability for Callbacks to handle failures

### CODE REVIEW

File: picasso/src/main/java/com/squareup/picasso/result/HTTPFailure.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
+package com.squareup.picasso.result;
+
+import com.squareup.picasso.Downloader.ResponseException;
+
+public class HttpFailure extends Failure {
```
Comment (seanfredericksen) [inline picasso/src/main/java/com/squareup/picasso/result/HTTPFailure.java]: File is named HTTPFailure rather than HttpFailure

Comment (abergfeld) [inline picasso/src/main/java/com/squareup/picasso/result/HTTPFailure.java]: Facepalm. I had some issues with Git not recognizing the file name change. Thought I had it fixed.

File: picasso/src/main/java/com/squareup/picasso/Dispatcher.java [hunk 2]
```diff
@@ -317,7 +321,7 @@ void performResumeTag(Object tag) {
     }
   }
 
-  void performRetry(BitmapHunter hunter) {
+  void performRetry(final BitmapHunter hunter) {
```
Comment (seanfredericksen) [inline picasso/src/main/java/com/squareup/picasso/Dispatcher.java]: Why did this need to change to final?

Comment (abergfeld) [inline picasso/src/main/java/com/squareup/picasso/Dispatcher.java]: I was using a visitor in this method at one point, this just didn't get cleaned up when I removed it.

**Comment by JohnKilcoyne** [comment]
:+1:

**Pull request merged**